backend/app/Http/Controllers/User/LessonController.php

```diff
@@ -24,42 +27,62 @@ public function store(Request $request)
 
         $temp = [];
         $new_answers = [];
-        foreach($answers['yourAnswer'] as $key => $value) 
+        foreach($answers as $value) 
         {
             if ($value) 
             {
                 $temp['lesson_id'] = $new_lesson->id;
-                $temp['question_id'] = $key;
-                $temp['choice_id'] = $value;
+                $temp['question_id'] = $value['question_id'];
+                $temp['choice_id'] = $value['choice_id'];
+                $temp['is_correct'] = $value['is_correct'];
                 $new_answers[] = $temp;
             }
         }
 
-        $answer_results = Answer::insert($new_answers);
+        Answer::insert($new_answers);
 
-        $activity_logs = ActivityLog::create([
+        ActivityLog::create([
             'user_id' => Auth::user()->id,
             'activity_id' => $new_lesson->id,
             'activity_type' => 'Lesson'
         ]);
 
         return response()->json([
-            'lessons' => $new_lesson,
-            'answers' => $answer_results,
-            'activity_logs' => $activity_logs
+            'lesson_id' => $new_lesson->id, 
+            'message' => 'You successfully submitted you answer!'
         ]);
     }
 
-    public function show($id)
+    public function show(Lesson $lesson)
     {
-        $lessons = Lesson::with([
-            'answers' => [
-                'question' => [
-                    'category'
-                ]
+        $get_category_title = Category::findOrFail($lesson->category_id)->title;
+        $count_question = count($lesson->answers);
+
+        $get_correct_answer = [];
+        $question_and_answer = [];
+        foreach ($lesson->answers as $value)
+        {
+            if ($value['is_correct'])
+            {
+                $get_correct_answer[] = $value['is_correct'];
+            }
+
+            $question_and_answer[] = [
+                'question_title' => Question::findOrFail((string)$value['question_id'])->value,
+                'is_correct' => $value['is_correct'],
+                'your_answer' => Choice::findOrFail((string)$value['choice_id'])->value
+            ];
+        }
+        
+        return response()->json([
+            'data' => [
+                'name' => Auth::user()->id === $lesson->user_id ? 'Your' : User::findOrFail($lesson->user_id)->name,
+                'get_category_title' => $get_category_title,
+                'get_question_and_answer' => $question_and_answer,
+                'count_question' => $count_question,
+                'count_correct_answer' => count($get_correct_answer)
```

Comment:
can this be an array of objects? `[{ }]`

Reply:
That's array of and object structure in laravel which is convertibly to [{}]
![image](https://user-images.githubusercontent.com/108642414/187358725-330df92a-d318-4508-9f23-4e6581bc73d2.png)